### PR TITLE
Users next page button fix

### LINKED ISSUE #364: Next page button in users/assignment view don't take into account search criteria
In the Users view and in the Assignment view, the next button don't take into account search criteria.

See the problem below (for the Users view):

cNyOEOR9W8

### PR DESCRIPTION
**Pull request by Asymmentric**
#364 Modified commit to include the changes which were made only in the necessary files. #379

### CODE REVIEW

**Comment by Asymmentric** [comment]
@NikhilM98 . can you please go through it once.

**Comment by llaske** [comment]
@Asymmentric could you close one of the two PR?

**Comment by Asymmentric** [comment]
Sure @llaske . I have closed the previous PR.

File: api/controller/users.js [hunk 1]
```diff
@@ -392,7 +391,7 @@ exports.findAll = function(req, res) {
 	query = addQuery('name', req.query, query);
 	query = addQuery('language', req.query, query);
 	query = addQuery('role', req.query, query, 'student');
-	query = addQuery('q', req.query, query);
+	query=addQuery('q',req.query,query)
```
Comment (llaske) [inline api/controller/users.js]: Avoid changing only indentation

File: api/controller/users.js [hunk 2]
```diff
@@ -470,7 +469,6 @@ function formPaginatedUrl(route, params, offset, limit) {
 
 //get all users
 exports.getAllUsers = function(query, options, callback) {
-
```
Comment (llaske) [inline api/controller/users.js]: Avoid changing only indentation

File: api/controller/users.js [hunk 3]
```diff
@@ -502,11 +500,9 @@ exports.getAllUsers = function(query, options, callback) {
 				}
 			}
 		];
-
```
Comment (llaske) [inline api/controller/users.js]: Avoid changing only indentation

File: api/controller/users.js [hunk 4]
```diff
@@ -502,11 +500,9 @@ exports.getAllUsers = function(query, options, callback) {
...
 		if (options.enableSecret == true) {
 			conf[1]["$project"]["uniqueSecret"] = 1;
 		}
-
```
Comment (llaske) [inline api/controller/users.js]: Avoid changing only indentation

File: api/controller/users.js [hunk 5]
```diff
@@ -520,8 +516,7 @@ exports.getAllUsers = function(query, options, callback) {
 				};
 			}
 		}
-
-		collection.aggregate(conf, function(err, users) {
+		collection.aggregate(conf, async function(err, users) {
```
Comment (llaske) [inline api/controller/users.js]: This not suppose to be async

File: api/controller/users.js [hunk 6]
```diff
@@ -559,11 +554,13 @@ function addQuery(filter, params, query, default_val) {
...
 	query = query || {};
-
 	//validate
 	if (typeof params[filter] != "undefined" && typeof params[filter] === "string") {
 
-		if (filter == 'q') {
+		// if (filter == 'q') {
+		if(filter=='q'){
+		
+		if(filter=='q'){
```
Comment (llaske) [inline api/controller/users.js]: I don't understand this double comparison

**Review by llaske — CHANGES_REQUESTED** [review]
It doesn't work, there is a fatal error.

image

Plus I've got doubt regarding your fix.

**Comment by Asymmentric** [comment]
hey @llaske  There was some issue with my git in the previous commits. Now its fixed.

File: api/controller/users.js [hunk 7]
```diff
@@ -45,8 +45,7 @@ exports.init = function(settings, database) {
  * @apiParam {String} id Unique user id
  * 
  * @apiSuccess {String} _id Unique user id
- * @apiSuccess {String} name Unique user name
- * @apiSuccess {String} role User role (student or admin)
+ * @apiSuccessearchs {String} role User role (student or admin)
```
Comment (llaske) [inline api/controller/users.js]: It's a typo, there is no reason to change the comment here

File: api/controller/users.js (hunk 1 above)
Comment (llaske) [reply api/controller/users.js]: Should be fixed

File: api/controller/users.js (hunk 2 above)
Comment (llaske) [reply api/controller/users.js]: Should be fixed

File: api/controller/users.js (hunk 3 above)
Comment (llaske) [reply api/controller/users.js]: Should be fixed

File: api/controller/users.js (hunk 5 above)
Comment (llaske) [reply api/controller/users.js]: Should be fixed

File: api/controller/users.js [hunk 8]
```diff
@@ -559,11 +554,10 @@ function addQuery(filter, params, query, default_val) {
 
 	//check default case
 	query = query || {};
-
```
Comment (llaske) [inline api/controller/users.js]: Should be fixed

File: api/controller/users.js [hunk 9]
```diff
@@ -559,11 +554,10 @@ function addQuery(filter, params, query, default_val) {
...
 	//validate
 	if (typeof params[filter] != "undefined" && typeof params[filter] === "string") {
-
-		if (filter == 'q') {
+
```
Comment (llaske) [inline api/controller/users.js]: Should be fixed

File: dashboard/controller/users/index.js [hunk 10]
```diff
@@ -71,7 +70,7 @@ exports.index = function(req, res) {
 
 				// get classrooms list
 				getClassrooms(req, function(classrooms){
-	
+
```
Comment (llaske) [inline dashboard/controller/users/index.js]: Should be fixed

File: dashboard/controller/users/searchUser.js [hunk 11]
```diff
@@ -15,8 +15,8 @@ module.exports = function searchUser(req, res) {
 	};
 
 	//get query params
-	if (req.query.q != '') {
-		query['q'] = req.query.q;
+	if(req.query.q != '') {
+		query['q']=req.query.q;
```
Comment (llaske) [inline dashboard/controller/users/searchUser.js]: Should be fixed

File: dashboard/controller/users/searchUser.js [hunk 12]
```diff
@@ -33,7 +33,7 @@ module.exports = function searchUser(req, res) {
 	if(req.query.sort !=''){
 		query['sort'] = req.query.sort;
 	}
-
+
```
Comment (llaske) [inline dashboard/controller/users/searchUser.js]: Should be fixed

**Review by llaske — CHANGES_REQUESTED** [review]
It works now.
Please fix the indentation since for maintenance reason we should be able to quickly identify lines changed.
Also the PR should be done on dev branch.

**Pull request closed without merging**